pkg/validation/validation.go

```diff
@@ -0,0 +1,51 @@
+package validation
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const ErrorNotFoundAgent = "unsupported fence agent %s:"
+
+type validateAgentExistence struct {
+	isFileExist func(string) (fs.FileInfo, error)
+}
+
+type AgentValidator interface {
+	ValidateAgentName(agent string) (admission.Warnings, error)
+}
+
+func NewAgentValidator(fileGetter func(string) (fs.FileInfo, error)) AgentValidator {
+	return &validateAgentExistence{isFileExist: fileGetter}
+}
+
+func (vfe *validateAgentExistence) ValidateAgentName(agent string) (admission.Warnings, error) {
+	exist, err := vfe.isAgentSupported(agent)
+	if !exist {
+		return admission.Warnings{"Check the agent spec field"}, fmt.Errorf(ErrorNotFoundAgent+" %w", agent, err)
```

Comment:
- the validator should not care about webhook specifics IMHO
- why adding a warning? The error should contain a clear message about what's wrong. IMHO warnings should be about issues which don't fail validation but e.g. can be improved... :thinking: